src/main/java/wromaciej/hvac_sim/solver/externals/ChannelSolver.java

```diff
@@ -0,0 +1,121 @@
+        return new Junction(parametersWithDirections, junctionSolver);
+    }
+
+    public Junction channelToEnergyJunction(Channel toSolve) {
+        List<ParameterWithDirection> parametersWithDirections = new ArrayList<>();
+        ParameterWithDirection specificEnthalpyDifference;
+        Parameter<SpecificEnthalpy> inletEnthalpyDefinition = new Parameter(ParameterType.SPECIFIC_ENTHALPY, SI.JOULE.divide(SI.KILOGRAM));
+        Parameter<SpecificEnthalpy> outletEnthalpyDefinition = new Parameter(ParameterType.SPECIFIC_ENTHALPY, SI.JOULE.divide(SI.KILOGRAM));
+        if (toSolve.getInletStream().getSpecificParameters().getSpecificEnthalpy().isDefined()) {
+            inletEnthalpyDefinition = toSolve.getInletStream().getSpecificParameters().getSpecificEnthalpy();
+        } else {
+            toSolve.getInletStream().getSpecificParameters().getMatterDefinition().addParameter(inletEnthalpyDefinition);
+        }
+        if (toSolve.getOutletStream().getSpecificParameters().getSpecificEnthalpy().isDefined()) {
+            outletEnthalpyDefinition = toSolve.getOutletStream().getSpecificParameters().getSpecificEnthalpy();
+        } else {
+            toSolve.getOutletStream().getSpecificParameters().getMatterDefinition().addParameter(outletEnthalpyDefinition);
+        }
+
+        if (toSolve.getHeatFlow().getParameter().isDefined()) {
+            toSolve.calculateSpecificEnthalpyDifferenceFromHeatFlow();
+        }
+        specificEnthalpyDifference = toSolve.getSpecificEnthalpyDifference();
+
+        parametersWithDirections.add(new ParameterWithDirection(inletEnthalpyDefinition, BondDirection.INLET));
+        parametersWithDirections.add(new ParameterWithDirection(outletEnthalpyDefinition, BondDirection.OUTLET));
+        parametersWithDirections.add(specificEnthalpyDifference);
+        return new Junction(parametersWithDirections, junctionSolver);
+
+    }
+
+
+    public SolverResult solveMassFlows(Channel channelToSolve) {
+        return junctionSolver.solve(channelToMassFlowJunction(channelToSolve));
+    }
+
+    public SolverResult solvePressures(Channel channelToSolve) {
+        return junctionSolver.solve(channelToPressureJunction(channelToSolve));
+    }
+
+    public SolverResult solveEnergy(Channel channelToSolve) {
+        SolverResult energySolverResult = junctionSolver.solve(channelToEnergyJunction(channelToSolve));
+        channelToSolve.calculateHeatFlowFromSpecificEnthalpyDifference();
+        return energySolverResult;
+    }
+
+
+    @Override
+    public SolverResult solve(Channel<? extends MatterStream> channelToSolve) {
+        SolverResultType solverResultType;
+        SolverResult massFlowSolverResult = solveMassFlows(channelToSolve);
+        SolverResult pressuresSolverResult = solvePressures(channelToSolve);
+        SolverResult energySolverResult = solveEnergy(channelToSolve);
+
+        if ((massFlowSolverResult.getResultType() == SOLVED)
```

Comment:
if you extracted the condition to method it would look like : 
```java
if(allResultsSolved(massFlowSolverResult, pressuresSolverResult, energySolverResult))
   solverResultType = SOLVED;
else solverResultType = SolverResultType.NOT_SOLVED_NODATA;
```
additionally you could extract that whole code to method and have like 
```java
SolverResultType solverResultType = resultTypeOf(massFlowSolverResult, pressuresSolverResult, energySolverResult)
```

hint: use method signature like 
```java
private SolverResultType resultTypeOf(SolverResult... results)
```